src/dmd/mtype.d

```diff
@@ -6732,8 +6735,17 @@ extern (C++) final class TypeNoreturn : Type
         //printf("TypeNoreturn::implicitConvTo(this=%p, to=%p)\n", this, to);
         //printf("from: %s\n", toChars());
         //printf("to  : %s\n", to.toChars());
-        MATCH m = Type.implicitConvTo(to);
-        return (m == MATCH.exact) ? MATCH.exact : MATCH.convert;
+        if (this.equals(to))
```

Comment:
`equals` does not take into account type qualifiers. Maybe it would be better to use `equivalent` here?

Reply:
Added a second check for different qualifiers